data/data-pipeline/data_pipeline/tests/conftest.py

```diff
@@ -0,0 +1,33 @@
+import os
+
+import pytest
+
+from data_pipeline.config import settings
+from data_pipeline.etl.base import ExtractTransformLoad
+
+TMP_DIR = settings.APP_ROOT / "data" / "tmp" / "tests"
+
+
+@pytest.fixture(scope="session")
+def mock_paths(tmp_path_factory) -> tuple:
+    """Creates new DATA_PATH and TMP_PATH that point to a temporary local
+    file structure that can be used to mock data folder during testing
+    """
+    # sets location of the temp directory inside the national_risk_index folder
+    os.environ["PYTEST_DEBUG_TEMPROOT"] = str(TMP_DIR)
+    TMP_DIR.mkdir(parents=True, exist_ok=True)
+    # creates DATA_PATH and TMP_PATH directories in temp directory
+    data_path = tmp_path_factory.mktemp("data", numbered=False)
+    tmp_path = data_path / "tmp"
+    tmp_path.mkdir()
+    return data_path, tmp_path
+
+
+@pytest.fixture
+def mock_etl(monkeypatch, mock_paths) -> None:
+    """Creates a mock version of the base ExtractTransformLoad class and resets
+    global the variables for DATA_PATH and TMP_PATH to the local mock_paths
+    """
+    data_path, tmp_path = mock_paths
+    monkeypatch.setattr(ExtractTransformLoad, "DATA_PATH", data_path)
+    monkeypatch.setattr(ExtractTransformLoad, "TMP_PATH", tmp_path)
```

Comment:
Thoughts on using env variables for paths rather than monkeypatching and setting the attributes? It seems to me simpler and requiring less code. 

I saw this pattern in @NatHillardUSDS unit test PR as well and now I'm wondering if this is just python convention; in my experience on Ruby projects using env vars has always been preferable to achieve this sort of thing.

Reply:
Hmm I have used this approach on a number of projects as well, and it's especially elegant with the `dynaconf` package since you can set variables either as environment variables or in a `settings.toml` 

Do you have a link to that line in Nate's PR? I'm currently having trouble envisioning how environment variables would allow us to circumvent the use of monkeypatching since the paths used in testing are different than those used when we're running the code live, which is one of the use cases that pytest suggests using monkeypatching for: https://docs.pytest.org/en/6.2.x/monkeypatch.html

Sorry if I'm missing the obvious workaround here!

Reply:
Also if we were to adopt use of environment variables instead of setting class attributes, I'd probably recommend scoping that into the ETL checks issue that I'm working on currently since it would require some refactoring of all of the ETL classes that inherit from the base `ExtractTransformLoad` class

Reply:
@switzersc-usds Ahhh I see your note [here](https://github.com/usds/justice40-tool/pull/509/files#r703030402) I do think this could be an elegant adjustment if we have separate environments in the `settings.toml` for `[development]` and `[testing]` 

I like the idea of setting `DATA_PATH` and `TMP_PATH` as config variables in the settings. There might be some implementation quirks involved in moving to that approach, but nothing we couldn't work through. That being said, it seems like something worth tackling in #617 or #576 to avoid scope creep for this particular PR.

Reply:
Discussed this a bit with Nat and Jorge when reviewing his PR and it looks like we need to refactor our use of dynaconf settings so that we set path names there (which will function essentially as env vars). We can proceed with this method for now and the next time we add some new tests we'll do that refactor.